### PR TITLE
Refactor DataCleaner and DataTransformer

### PR DESCRIPTION
**Pull request by btseytlin**
Resolves:
https://github.com/mindsdb/mindsdb_native/issues/80
https://github.com/mindsdb/mindsdb_native/issues/52

### CODE REVIEW

File: mindsdb_native/libs/backends/lightwood.py [hunk 1]
```diff
@@ -87,10 +87,7 @@ def _create_lightwood_config(self):
         config['input_features'] = []
         config['output_features'] = []
 
-        for col_name in self.transaction.input_data.columns or col_name not in self.transaction.lmd['stats_v2']:
-            if col_name in self.transaction.lmd['columns_to_ignore']:
-                continue
-
+        for col_name in self.transaction.input_data.columns:
```
Comment (maximlopin) [inline mindsdb_native/libs/backends/lightwood.py:90]: shouldn't `columns_to_ignore` be skipped?

Comment (btseytlin) [reply mindsdb_native/libs/backends/lightwood.py:90]: They are dropped from input data before that

**Comment by George3d6** [comment]
Added my thoughts on this here: https://www.notion.so/mindsdb/Data-Cleaner-Transformer-redesign-evaluation-717af604d6b44606a31c024f6f9a4119

TL;DR

What's happening in the PR makes sense, in this form though it touches a few too many things and it would break some functionality, so I'd rather modify it slightly and merge it later.

Will close for now and re-open in the future when we have time to address those things.
